python/tank/util/shotgun/download.py

```diff
@@ -182,7 +184,10 @@ def download_and_unpack_attachment(sg, attachment_id, target, retries=5):
 
             log.debug("Unpacking %s bytes to %s..." % (file_size, target))
             filesystem.ensure_folder_exists(target)
-            unzip_file(zip_tmp, target)
+            try:
+                unzip_file(zip_tmp, target)
+            except zipfile.BadZipfile:
+                invalid_zip_file = True
```

Comment:
attachment download now reports a better error message when an attachment isn't a zip file.

Reply:
I think we should call `log.exception` with `BadZipFile` errors. This would be great for support. Maybe there's something in that exception that could help us debug the issue? Truncated file vs corrupted block vs something else?

Reply:
So the main reason badzipfile is raised is because whatever was in shotgun was not a zip file. The exception raised in this case (further down in the code is)

```
raise ShotgunAttachmentDownloadError("Shotgun attachment with id %s is not a zip file!" % attachment_id)
```

i don't think the exception itself really provides any more info than that so i am not sure i see how logging the exception could help support in this case. 

It can be bad to raise and log - since the calling code may want to do its own logging - or not even consider it an error that needs logging - instead it catches the exception and handles it gracefully.